packages/bundler/src/BundlerServer.ts

```diff
@@ -107,27 +124,70 @@ export class BundlerServer {
     res.send(`Account-Abstraction Bundler v.${erc4337RuntimeVersion}. please use "/rpc"`)
   }
 
-  async rpc (req: Request, res: Response): Promise<void> {
-    let resContent: any
-    if (Array.isArray(req.body)) {
-      resContent = []
-      for (const reqItem of req.body) {
-        resContent.push(await this.handleRpc(reqItem))
+  // TODO: I don't see how to elegantly combine express callbacks with classes so I ended up with this spaghetti.
+  //  This is temporary and probably should not be merged like that, we need to simplify the flow.
+  getRpc (handleRpc: any): any {
+    const rpc = async (req: Request, res: Response): Promise<void> => {
+      let resContent: any
+      if (Array.isArray(req.body)) {
+        resContent = []
+        for (const reqItem of req.body) {
+          resContent.push(await handleRpc(reqItem))
+        }
+      } else {
+        resContent = await handleRpc(req.body)
+      }
+
+      try {
+        res.send(resContent)
+      } catch (err: any) {
+        const error = {
+          message: err.message,
+          data: err.data,
+          code: err.code
+        }
+        this.log('failed: ', 'rpc::res.send()', 'error:', JSON.stringify(error))
       }
-    } else {
-      resContent = await this.handleRpc(req.body)
     }
+    return rpc.bind(this)
+  }
 
-    try {
-      res.send(resContent)
-    } catch (err: any) {
+  // TODO: deduplicate!
+  async handleRpcPublic (reqItem: any): Promise<any> {
+    const { method, jsonrpc, id } = reqItem
+
+    if (method === 'eth_getRip7560Bundle') {
       const error = {
-        message: err.message,
-        data: err.data,
-        code: err.code
+        message: `requested RPC method (${method as string}) is not available`,
+        data: '',
+        code: ValidationErrors.InvalidRequest
+      }
+      return {
+        jsonrpc,
+        id,
+        error
       }
-      this.log('failed: ', 'rpc::res.send()', 'error:', JSON.stringify(error))
     }
+    return await this.handleRpc(reqItem)
+  }
+
+  // TODO: deduplicate!
+  async handleRpcPrivate (reqItem: any): Promise<any> {
```

Comment:
I don't understand this private/public distinction then - They just sit on different ports, but there are no restrictions on who can call the "private" api. Are we only planning to block requests on the "devops" side and not in code?
if so, shouldn't we just name it "blockBuilderRpcHandler" and "clientRpcHandler" since there is not private/public distinction in code? we basically rely on the cloud configuration to make sense of these names

Reply:
I can rename it but I think in the future we may end up creating other use cases for an API that is only available "internally".
I think as long as it is trivial to prevent access to the "private" API with i.e. basic docker configuration there is no point in writing any access control code.